src/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/SignerInfo.cs

```diff
@@ -883,32 +813,9 @@ private static (int, int) FindAttributeLocation(AttributeAsn[] attributes, AsnEn
 
         private static void RemoveAttributeValueWithoutIndexChecking(ref AttributeAsn modifiedAttr, int removeValueIndex)
         {
-            // Using BER rules to avoid resorting
-            using (AsnWriter writer = new AsnWriter(AsnEncodingRules.BER))
+            if (removeValueIndex < modifiedAttr.AttrValues.Length)
```

Comment:
is this even possible not to be true? if not I'd change it into assert

Reply:
Good question, I wrote it that way because that's how the original code behaved. I will check the callers.

Reply:
You are right. I will just inline the one line to both of the callers.